### ISSUE TITLE
Research what method to generate SCIG's works best for us

### ISSUE THREAD

**Issue by b21oscpe** [HGustavs/LenaSYS#13555]
Find an applicable method of generating a state chart intermediate graph that we can use in a later issue #13554

**Comment by a21nibah** [comment]
is the goal to be able to generate a SCIG within the diagram tool or can external software be used?

**Comment by a21nibah** [comment]
The first method that i found involves using a software called whitestar uml. You would have to draw your diagram and then generate the sig using the software. The issue is that in order to transition this to code you will have to get the data from the XMI file. From what i have read you can fetch data like you would with a xml file. I do not know the process of generating the scig within the software since the software is not accessible.
https://sourceforge.net/projects/whitestaruml/
I’ll keep looking for alternatives.

**Comment by b21oscpe** [comment]
> is the goal to be able to generate a SCIG within the diagram tool or can external software be used?

Preferably within the diagram tool however external software may be implemented if it proves easier, we do not want to reinvent the wheel

**Comment by a21nibah** [comment]
just to clerify are we supposed to generate a SCIG from an existing graph correct? that is we have a diagram and from that we have to generate a SCIG based on that diagram?

**Comment by b21matfi** [comment]
Are we looking after method like Lucidchart or? Need a better explanation

**Comment by b21oscpe** [comment]
> just to clerify are we supposed to generate a SCIG from an existing graph correct? that is we have a diagram and from that we have to generate a SCIG based on that diagram?

Exactly, with the data that we are able to extract from a state diagram we should be able to generate a SCIG. Your task is not to actually implement it yet but to research what would be the most appropriate way for us to make this possible.

**Comment by b21oscpe** [comment]
I suggest you create a google document where you can all share your conclusions

**Comment by a21nibah** [comment]
Alright. I have spent time trying to figure out how you actually can generate a scig from a state diagram. Because a scig is a more detailed form of a state diagram where we have a directed graph that shows in which order a transition occurs the issue was how we correctly infer the transition order from the state diagram. For example, what happends if you have a state with multiple transitions? How can you know from the state diagram where each transition is supposed to go where?
In other words trying to translate a state diagram to a SCIG is not really possible I think.
But after talking to the project leader today the conclusion we reached was that we actually do not really need a scig. It appears that henrik is ok with us only printing the state diagram. And based on previous research I do not believe that the algorithm we will use really needs a scig since it is only testing to see if we can go through each path in the diagram.
I would suggest looking att issue #13437 because there is already some code that is not working currently that should print out the correct connections between each node in the state diagram.
